### PR TITLE
Feat/custom entity detail view

### PR DESCRIPTION
**Pull request by connoratrug**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] Conventional commits (squash if needed)
- [ ] No warnings during install
- [ ] Updated javascript typing

### CODE REVIEW

**Comment by connoratrug** [comment]
[#7230 (sub task) custom entity detail view](https://trello.com/c/6sDPSkvw/447-7230-sub-task-custom-entity-detail-view)

File: packages/data-explorer/src/components/dataView/DefaultCardContent.vue [hunk 1]
```diff
@@ -41,9 +41,9 @@
             <font-awesome-icon icon="chevron-up" v-if="cardState==='open'"></font-awesome-icon>
             <font-awesome-icon icon="chevron-right" v-if="cardState==='closed'"></font-awesome-icon> {{ expandBtnText }}
           </button>
-          <a :href="detailLink" class="btn btn-outline-info btn-sm mg-info-btn">
+          <router-link class="btn btn-outline-info btn-sm mg-info-btn" role="button" :to="{ name: 'entity-detail', params: { entityType: dataTable, entity: dataId}}">
```
Comment (jelmerveen) [inline packages/data-explorer/src/components/dataView/DefaultCardContent.vue:44]: multiline & space between }}

File: packages/data-explorer/src/components/dataView/DefaultEntityDetail.vue [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<template>
+    <ul class="list-group list-group-flush">
+      <li class="list-group-item" v-for="(value, propertyName, index) in record" :key="index">
+        <h5>{{propertyName}}</h5>
+        <template v-if="typeof value !== 'object'">
+           <p>{{value}}</p>
+        </template>
+        <template v-else-if="Array.isArray(value)">
+          {{value.map(v => v.label).join(', ')}}
```
Comment (jelmerveen) [inline packages/data-explorer/src/components/dataView/DefaultEntityDetail.vue:9]: missing space {{ }}

File: packages/data-explorer/src/components/dataView/DefaultEntityDetail.vue [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        </template>
+        <template v-else>
+          <ul class="list-group list-group-flush">
+            <li class="list-group-item" v-for="(value, propertyName, refIndex) in value" :key="refIndex">
+              <h5>{{propertyName}}</h5>
```
Comment (jelmerveen) [inline packages/data-explorer/src/components/dataView/DefaultEntityDetail.vue:14]: missing space {{ }}

File: packages/data-explorer/src/components/dataView/DefaultEntityDetail.vue [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+              <p>{{value}}</p>
```
Comment (jelmerveen) [inline packages/data-explorer/src/components/dataView/DefaultEntityDetail.vue:15]: missing space {{ }}

File: packages/data-explorer/src/components/dataView/TableRow.vue [hunk 5]
```diff
@@ -12,14 +12,13 @@
         :to="{ name: 'de-edit', params: { entity: tableName, dataRowId: id}, query: {}}">
```
Comment (jelmerveen) [inline packages/data-explorer/src/components/dataView/TableRow.vue:12]: missing space }}

File: packages/data-explorer/src/components/dataView/TableRow.vue [hunk 6]
```diff
@@ -12,14 +12,13 @@
...
         <font-awesome-icon icon="edit"></font-awesome-icon>
       </router-link>
-      <a
+      <router-link
+        class="btn btn-sm text-secondary" role="button"
+        :to="{ name: 'entity-detail', params: { entityType: tableName, entity: id}}"
```
Comment (jelmerveen) [inline packages/data-explorer/src/components/dataView/TableRow.vue:17]: missing space }}

File: packages/data-explorer/src/views/EntityDetail.vue [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+            <router-link id="headItemTooltipID" :to="{ name: 'de-view', params: { entity: entityType} }">
+            <b-tooltip v-if="metaData.description" placement="bottom" target="headItemTooltipID" triggers="hover">
+              {{ metaData.description }}
+            </b-tooltip>
+            {{ metaData.label }}
+            </router-link>
+          </li>
+          <li class="breadcrumb-item" aria-current="page">
+            <span>
+               {{record[metaData.idAttribute.name]}}
```
Comment (jelmerveen) [inline packages/data-explorer/src/views/EntityDetail.vue:18]: missing space {{ }}

File: packages/data-explorer/src/views/EntityDetail.vue [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    </div>
+
+    <div class="btn-group float-right pr-3 mr-3" role="group" aria-label="row actions group">
+
+      <router-link
+        v-if="hasAddRights"
+        class="btn btn-outline-secondary"
+        v-b-tooltip.hover.bottom
+        :title="$t('dataexplorer_add_entity_btn_tooltip')"
+        :to="{ name: 'de-create', params: { entity: entityType }}"
```
Comment (jelmerveen) [inline packages/data-explorer/src/views/EntityDetail.vue:32]: missing space }}

File: packages/data-explorer/src/views/EntityDetail.vue [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+      >
+        <font-awesome-icon icon="plus-square"></font-awesome-icon>
+      </router-link>
+
+      <router-link
+        v-if="hasEditRights"
+        class="btn btn-outline-secondary"
+        v-b-tooltip.hover.bottom
+        :title="$t('dataexplorer_row_action_edit_btn_tooltip')"
+        :to="{ name: 'de-edit', params: { entity: entityType, dataRowId: entity}}"
```
Comment (jelmerveen) [inline packages/data-explorer/src/views/EntityDetail.vue:42]: missing space }}

File: packages/data-explorer/tests/unit/components/dataView/CustomEntityDetail.spec.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+      meta1: 'meta 1',
+      meta2: 'meta 2'
+    },
+    template: '    <h1>Hello MOLGENIS!</h1>    '
+  }
+  const wrapper = shallowMount(CustomEntityDetail, { propsData })
+  it('exists', () => {
+    expect(wrapper.exists()).toBeTruthy()
+  })
+  it('should set the template', () => {
```
Comment (jelmerveen) [inline packages/data-explorer/tests/unit/components/dataView/CustomEntityDetail.spec.ts:21]: should it also strip whitespace? Your template obviously has whitespace padding left and right.

File: packages/data-explorer/tests/unit/views/EntityDetail.spec.ts [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    await wrapper.vm.$nextTick // settings
+    await wrapper.vm.$nextTick // data
+  })
+
+  describe('delete', () => {
+    it('should  call the delete actions when the request is confirmed', async () => {
+      actions.deleteRow.mockResolvedValueOnce(true)
+      mocks.$bvModal.msgBoxConfirm.mockResolvedValueOnce(true)
+      await wrapper.find('.delete-btn').trigger('click')
+      await wrapper.vm.$nextTick()
```
Comment (jelmerveen) [inline packages/data-explorer/tests/unit/views/EntityDetail.spec.ts:84]: do we need to wait for nextTick after the click?

File: packages/data-explorer/tests/unit/views/EntityDetail.spec.ts [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+      expect(actions.deleteRow).toHaveBeenCalled()
+    })
+    it('should not call the delete actions when the request is canceled', async () => {
+      mocks.$bvModal.msgBoxConfirm.mockResolvedValueOnce(false)
+      await wrapper.find('.delete-btn').trigger('click')
+      await wrapper.vm.$nextTick()
```
Comment (jelmerveen) [inline packages/data-explorer/tests/unit/views/EntityDetail.spec.ts:90]: do we need nextTick?

Comment (jelmerveen) [reply packages/data-explorer/tests/unit/views/EntityDetail.spec.ts:90]: just tried it, and yes this is needed. Could you maybe add a comment what we are waiting for?

**Comment by connoratrug** [comment]
outdated

**Pull request closed without merging**